compiler/ballerina-lang/src/main/java/io/ballerina/projects/CompilationOptions.java

```diff
@@ -93,4 +100,11 @@ private boolean toBooleanDefaultIfNull(Boolean bool) {
         }
         return bool;
     }
+
+    private String toStringDefaultIfNull(String value) {
+        if (value == null) {
+            return "k8s";
```

Comment:
Won't this try to generate the cloud artifacts even when the `--cloud` flag is not set?